addons/editor-extra-blocks/addon.json

```diff
@@ -0,0 +1,34 @@
+{
+  "name": "Reveal hidden editor blocks",
+  "description": "TODO",
+  "tags": ["editor", "codeEditor"],
+  "credits": [
+    {
+      "name": "Tacodiva",
+      "link": "https://github.com/Tacodiva"
+    }
+  ],
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["projects"]
+    }
+  ],
+  "settings": [
+    {
+      "name": "Show Counter blocks",
```

Comment:
Why do the counter blocks get its own setting? Is it because they're not very useful compared to the loops and when touching ones?

Reply:
Yeah, it was my opinion that they where kinda useless compared to the other blocks and there's a few of them so I thought it best to put them in their own setting.

Reply:
Kinda New To GitHub, So I'm Not Sure If I Should Say Stuff A Month Late, But I Will Say That Counter Blocks Have A Specifice Use Case. They Reset To 0 ANY-time A project Is Loaded, Reloaded, Remixed. I Use These In Every Project For A One-Time Intro That Changes If The Project Is Remixed. Somthing Not Possible With Variables. Yes, Specific. The Value Is The Only Value Not Saved In The Json.

Reply:
Yeah, I've seen it used that way before.

Reply:
Remix detector? 💀

Reply:
Yup. I Also Learned It's possible To Do The Same Thing With A Complicated Username Setup, But The Counter Blocks make it more Convenient... So IDK How These Blocks Should be handled For Scratch Addons.

Reply:
> Yup. I Also Learned It's possible To Do The Same Thing With A Complicated Username Setup, But The Counter Blocks make it more Convenient & Reliable... So IDK How These Blocks Should be handled For Scratch Addons.
> 
> It's More Like A On Project Start Detector, But Can be Used For Detecting Remixes.

But what would you do with a remix detector (other than blocking access to your project?)

Reply:
Sense We Arn't Really Allowed To Block Access, I Use it As A Extra Failsafe in Case If Somewon Uploads A Project & It Doesn't Say The Remix, But Should be.

Reply:
> Sense We Arn't Really Allowed To Block Access, I Use it As A Extra Failsafe in Case If Somewon Uploads A Project & It Doesn't Say The Remix, But Should be.

That's a little extreme, though... is that really a reason for Scratch Addons to support counter blocks?
> One Thing I'm Tring To Figure Out Is The Most Effective Way Of Detecting Wether They broke The Failsafe, & Most Effective Contraption. At That Point It Would Block Access To The Project.

That's too complicated, probably not realistically possible, and risks violating Scratch's TOS.

(Also, might I suggest that you not capitalize every word you write?)

Reply:
I guess it kinda would violate, but the remixer would be too.

IDK how Scratch Addons should handle these blocks. Even though I really want easier access to these blocks.

Counter blocks don't easily have a readying monitor, so they kinda also work as a "hidden" variable.

Reply:
> I guess it kinda would violate, but the remixer would be too.

To cite the adage: two wrongs don't make a right
> IDK how Scratch Addons should handle these blocks. Even though I really want easier access to these blocks.

Yeah. It's been somewhat contentious, but I generally support adding them.
> Counter blocks don't easily have a readying monitor, so they kinda also work as a "hidden" variable.

That's their main advantage. Sometimes I'll use them, but I always ask how it's better then a variable 🤷

Reply:
It would only really actually violate if the remixer violates. But yes thats true.

I Still Feel Like They Don't Need A Seprate Toggle, like the while block just saves 1 not, and the "loud?" block replaces "loudness > 10", while the counter blocks give new mechanics. As of now the hidden blocks just fulfill a niche, not nessecarly be great. the "for each [] in ()" block is probably the 1 exception.

Reply:
@KingKittyTurnip @BroJac5246 This conversation seems to be getting a bit sidetracked. I chose to hide the counter blocks by default because, unlike the other blocks, they don't have an 'intuitive' use. It's obvious what the 'while' block does, or the 'for each' or even '<loud ?>'. These have usecases that the average Scratcher can likely grasp very easily. But on a surface level, the counter is just a more restrictive variable with no real utility. Additionally, there are 3 different counter blocks which all take up valuable space in the 'control' category. 

I have the same opinion about the `all at once` block. It can be used for `all at once { delete this clone }` to continue script execution for the original sprite but not the clones, yet I don't think this use is obvious so it too is hidden by default.

All this being said, it's really not that hard to flip a switch in the settings menu. Unless someone else has something to add to this discussion, I stand by my decision.